src/atelier.cpp

```diff
@@ -0,0 +1,191 @@
+#include "atelier.h"
+#include <QCoreApplication>
+#include <QFile>
+#include <QEventLoop>
+#include <QTextStream>
+#include <QDir>
+#include <QTime>
+#include <KAtCore/seriallayer.h>
+
+Atelier::Atelier():
+    core(new AtCore(this)),
+    logFile(new QTemporaryFile(QDir::tempPath() + QStringLiteral("/AtCore_")))
+{
+    connect(core, &AtCore::printerStatusChanged, this, &Atelier::checkPrinterStatus);
+}
+
+Atelier::~Atelier()
+{
+    delete logFile;
+}
+
+void Atelier::startConnectProccess(QString port, QString baud)
+{
+    core->initFirmware(port, baud.toInt());
+    connect(core->serial(), &SerialLayer::receivedCommand, this, &Atelier::checkReceivedCommand);
+    connect(core->serial(), &SerialLayer::pushedCommand, this, &Atelier::checkPushedCommands);
+}
+
+void Atelier::checkPrinterStatus(PrinterStatus status)
+{
+   QString msg = QString::fromLatin1("Bed:%1/%2  Extruder:%3/%4")
+                      .arg(QString::number(status.bedTemp))
+                      .arg(QString::number(status.bedTargetTemp))
+                      .arg(QString::number(status.extruderTemp))
+                      .arg(QString::number(status.extruderTargetTemp));
+    emit updateTemperatures(status.bedTemp,status.extruderTemp);
+    addRLog(msg);
+}
+
+void Atelier::pausePrint()
```

Comment:
use AtCore::pause(const QString &pauseActions)
Do not set the PAUSE state directly. This will cause problems on resume